vcd/resource_vcd_network.go

```diff
@@ -162,22 +166,39 @@ func resourceVcdNetworkCreate(d *schema.ResourceData, meta interface{}) error {
 		return fmt.Errorf(errorRetrievingOrgAndVdc, err)
 	}
 
-	edgeGateway, err := vdc.FindEdgeGateway(d.Get("edge_gateway").(string))
-	if err != nil {
-		return fmt.Errorf(errorUnableToFindEdgeGateway, err)
+	// Collecting components
+	egName, ok := d.GetOk("edge_gateway")
+	var edgeGateway govcd.EdgeGateway
+	fenceMode := "natRouted"
+
+	edgeGatewayName := ""
+	if ok {
+		edgeGatewayName = egName.(string)
```

Comment:
I don't see usage 'edgeGatewayName' in code. Directly egName can be used.

Reply:
This code will disappear. Anyway, `egName` is `interface{}`, while `edgeGatewayName` is a string.